File: Assets/AltUnityTester/AltUnityServer/Commands/UnityCommands/SetTimeScale.cs
namespace Assets.AltUnityTester.AltUnityServer.Commands
{
    class SetTimeScale:Command
    {
        float timeScale;

        public SetTimeScale(float timeScale)
        {
            this.timeScale = timeScale;
        }

        public override string Execute()
        {
            AltUnityRunner._altUnityRunner.LogMessage("SetTimeScale to: " + timeScale);
            string response = AltUnityRunner._altUnityRunner.errorCouldNotPerformOperationMessage;
            UnityEngine.Time.timeScale = timeScale;
            response = "Ok";
            return response;
        }
    }
}

File: Bindings~/python/altunityrunner/commands/UnityCommands/set_time_scale.py
from loguru import logger

from altunityrunner.commands.base_command import BaseCommand


class SetTimeScale(BaseCommand):

    def __init__(self, connection, time_scale):
        super().__init__(connection, "setTimeScale")

        self.time_scale = time_scale

    @property
    def _parameters(self):
        parameters = super()._parameters
        parameters.update(**{
            "timeScale": str(self.time_scale),
        })

        return parameters

    def execute(self):
        logger.debug("Set time scale to: {}", self.time_scale)
        data = self.send()

        if (data == "Ok"):
            logger.debug("Time scale set to: {}", self.time_scale)
            return data

        return None
